shared/src/test/scala/kinesis4cats/producer/ProducerSpec.scala

```diff
@@ -33,64 +33,68 @@ import kinesis4cats.models.ShardId
+      raiseOnFailures: Boolean
+  ): SyncIO[FunFixture[MockProducer]] =
     ResourceFunFixture(
-      MockProducer(aggregate)
+      MockProducer(aggregate, raiseOnFailures)
     )
 
-  fixture(false).test("It should retry methods and eventually produce") {
-    producer =>
-      val record1 = Record(Array.fill(50)(1), "1")
-      val record2 = Record(Array.fill(50)(1), "2")
-      val record3 = Record(Array.fill(50)(1), "3")
-      val record4 = Record(Array.fill(50)(1), "4")
-      val record5 = Record(Array.fill(50)(1), "5")
-
-      val data = NonEmptyList.of(
-        record1,
-        record2,
-        record3,
-        record4,
-        record5
-      )
+  fixture(aggregate = false, raiseOnFailures = true).test(
+    "It should retry methods and eventually produce"
+  ) { producer =>
+    val record1 = Record(Array.fill(50)(1), "1")
+    val record2 = Record(Array.fill(50)(1), "2")
+    val record3 = Record(Array.fill(50)(1), "3")
+    val record4 = Record(Array.fill(50)(1), "4")
+    val record5 = Record(Array.fill(50)(1), "5")
 
-      val response1 = MockPutResponse(
-        NonEmptyList.one(record1),
-        List(record2, record3, record4, record5)
-      )
+    val data = NonEmptyList.of(
+      record1,
+      record2,
+      record3,
+      record4,
+      record5
+    )
 
-      val response2 = MockPutResponse(
-        NonEmptyList.one(record2),
-        List(record3, record4, record5)
-      )
+    val response1 = MockPutResponse(
+      NonEmptyList.one(record1),
+      List(record2, record3, record4, record5)
+    )
 
-      val response3 = MockPutResponse(
-        NonEmptyList.one(record3),
-        List(record4, record5)
-      )
+    val response2 = MockPutResponse(
+      NonEmptyList.one(record2),
+      List(record3, record4, record5)
+    )
 
-      val response4 = MockPutResponse(
-        NonEmptyList.one(record4),
-        List(record5)
-      )
+    val response3 = MockPutResponse(
+      NonEmptyList.one(record3),
+      List(record4, record5)
+    )
 
-      val response5 = MockPutResponse(
-        NonEmptyList.one(record5),
-        List.empty
-      )
+    val response4 = MockPutResponse(
+      NonEmptyList.one(record4),
+      List(record5)
+    )
 
-      val expected: Producer.Result[MockPutResponse] = Producer.Result(
-        List(response1, response2, response3, response4, response5),
-        Nil,
-        Nil
-      )
+    val response5 = MockPutResponse(
+      NonEmptyList.one(record5),
+      List.empty
+    )
 
-      producer.put(data).map { res =>
-        assert(res === expected, s"res: $res\nexp: $expected")
-      }
+    val expected: Producer.Result[MockPutResponse] = Producer.Result(
+      List(response1, response2, response3, response4, response5),
+      Nil,
+      Nil
+    )
+
+    producer.put(data).map { res =>
+      assert(res === expected, s"res: $res\nexp: $expected")
+    }
   }
 
-  fixture(true).test(
+  fixture(aggregate = true, raiseOnFailures = true).test(
```

Comment:
```suggestion
  fixture(aggregate = true, raiseOnFailures = false).test(
```

I think we want this as false, yes?

Reply:
First of all, really sorry for the super late reply 🙏🏻 

At first I was a bit confused because I noticed the test already had `raiseOnFailures = true`, so after changing it and running it locally, I didn’t understand why the behavior seemed unchanged: https://github.com/etspaceman/kinesis4cats/blob/8012752b92792917cf9f5a1be433f3b90cd64874/shared/src/test/scala/kinesis4cats/producer/ProducerSpec.scala#L238

After digging in again, I think I get what’s happening: in the test, we're using `Result[MockPutResponse]`, and each `MockPutResponse` includes both success and failure records inside it. So even when a response contains failed records, it still ends up in the `successful: List[MockPutResponse]`:
https://github.com/etspaceman/kinesis4cats/blob/8012752b92792917cf9f5a1be433f3b90cd64874/shared/src/test/scala/kinesis4cats/producer/ProducerSpec.scala#L135-L139

That means the failed list inside the `Result` itself is always empty, so the `raiseOnFailures` flag doesn’t actually trigger anything in this case.

Might make sense to address this properly in a follow-up PR, since it could impact the whole suite?

Reply:
Sure we can do a follow up PR